test/mocha/20-totp.js

```diff
@@ -0,0 +1,200 @@
+    });
+    it('should verify a valid token', async () => {
+      const totpToken = totp.generate(secret);
+      let err;
+      let result;
+      try {
+        result = await brAuthnToken.verify({
+          account: accountId,
+          actor,
+          type: 'totp',
+          totpToken,
+        });
+      } catch(e) {
+        err = e;
+      }
+      assertNoError(err);
+      should.exist(result);
+      result.should.eql({
+        id: accountId,
+        email: mockAccountEmail,
+        token: {type: 'totp', authenticationMethod: 'totp'}
+      });
+    });
+    it('should not verify an invalid token', async () => {
+      // an invalid random totpToken string
+      const totpToken = '91bc2849-2f43-4847-b5ad-01c0436b3a07';
+      let err;
+      let result;
+      try {
+        result = await brAuthnToken.verify({
+          account: accountId,
+          actor,
+          type: 'totp',
+          totpToken,
+        });
+      } catch(e) {
+        err = e;
+      }
+      assertNoError(err);
+      should.exist(result);
+      result.should.be.a('boolean');
+      result.should.be.false;
+    });
+  }); // end verify
+
+  describe('delete', () => {
+    const mockAccountEmail = 'alpha@example.com';
+    let secret;
+    let accountId;
+    let actor;
+    before(async () => {
+      await helpers.prepareDatabase(mockData);
+    });
+    // set an OTOP secret
+    before(async () => {
+      accountId = mockData.accounts[mockAccountEmail].account.id;
+      actor = await brAccount.getCapabilities({id: accountId});
+      ({secret} = await brAuthnToken.set({
+        account: accountId,
+        actor,
+        type: 'totp',
+      }));
+    });
+    it('should delete a secret', async () => {
+      let totpToken = totp.generate(secret);
+      let err;
+      let result;
+      try {
+        result = await brAuthnToken.verify({
+          account: accountId,
+          actor,
+          type: 'totp',
+          totpToken,
+        });
+      } catch(e) {
+        err = e;
+      }
+      assertNoError(err);
+      should.exist(result);
+      result.should.eql({
+        id: accountId,
+        email: mockAccountEmail,
+        token: {type: 'totp', authenticationMethod: 'totp'}
+      });
+
+      err = null;
+      result = null;
+      try {
+        result = await brAuthnToken.remove({
+          account: accountId,
+          actor,
+          type: 'totp',
+        });
+      } catch(e) {
+        err = e;
+      }
+      should.not.exist(err);
+
+      // attempt to validate the TOTP token again
+      totpToken = totp.generate(secret);
+      err = null;
+      result = null;
+      try {
+        result = await brAuthnToken.verify({
+          account: accountId,
+          actor,
+          type: 'totp',
+          totpToken,
+        });
+      } catch(e) {
+        err = e;
+      }
+      should.exist(err);
+      err.name.should.equal('NotFoundError');
+
+      // FIXME: SHOULD THE VERIFY API DO THIS INSTEAD?
```

Comment:
I'm going to file an issue so we can consider this later and we should leave it as is for now.